src/services/dynamo-database-service.ts

```diff
@@ -100,15 +100,29 @@ export class DynamoDatabaseService {
         ':ttl': { N: ttl.toString() },
         ':false': { BOOL: false },
       },
-      ConditionExpression: 'attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false',
+      ConditionExpression:
+        'attribute_exists(pk) AND (attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false)',
     };
     const command = new UpdateItemCommand(commandInput);
-    const response = await this.dynamoClient.send(command);
-    if (!response) {
-      const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+    try {
+      const response = await this.dynamoClient.send(command);
+      logger.info(`${LOGS_PREFIX_SENSITIVE_INFO} Account ${userId} marked as deleted.`);
+      if (!response) {
+        const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
+        logger.error(errorMessage);
+        logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+        return;
+      }
+      logAndPublishMetric(MetricNames.MARK_AS_DELETED_SUCCEEDED);
+      return response;
+    } catch (error: any) {
+      if (error.name != 'ConditionalCheckFailedException') {
+        const errorMessage = `${LOGS_PREFIX_SENSITIVE_INFO} Error updating item with pk ${userId}.`;
+        logger.error(errorMessage);
+        logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+        throw new Error('Error was not a Conditional Check Exception.'); //Therefore re-driving message back to the queue.
```

Comment:
```suggestion
        throw new Error('Error was not a Conditional Check Exception.', { error }); //Therefore re-driving message back to the queue.
```

Reply:
Unable to include error as an object here, as the error is unknown/ any and it has to be specified to be used here

Reply:
Hey Jess, I have tried this and you should be able to do:
```
catch (error: any) {
    logger.error("message", { error: error });
    if(!error.name || error.name == "something")
```
so that we do not loose the original error message